app/src/main/java/com/example/anfcodetest/MainPresenter.kt

```diff
@@ -1,103 +1,51 @@
 package com.example.anfcodetest
 
-import android.content.Intent
-import android.text.Html
-import android.text.method.LinkMovementMethod
-import android.view.View.GONE
-import com.squareup.picasso.Picasso
-import kotlinx.android.synthetic.main.card_row.view.*
-
-
-class MainPresenter : MainContract.Presenter {
-
-    override fun cardSetup(holder: CustomViewHolder, position: Int, cardList: List<Card>) {
-        val card = cardList[position]
-
-        //Check if there is a card image for the current card and remove the view if not
-        if (card.backgroundImage.isNullOrBlank()) {
-            holder.view.cardImage.visibility = GONE
-        } else {
-            Picasso.get().load(card.backgroundImage).into(holder.view.cardImage)
-        }
-
-        //Check if there is a title for the current card and remove the view if not
-        if (card.title.isNullOrEmpty()) {
```

Comment:
This is a great example of a place where `setTextAndVisibility` would be helpful - we only want to make this visible if we have text! Check out the extension functions below for this stuff.